test/composeWith.js

```diff
@@ -0,0 +1,47 @@
+var R = require('..');
+var eq = require('./shared/eq');
+
+
+describe('composeWith', function() {
+
+  it('performs right-to-left function composition with function applying', function() {
+    //  f :: (String, Number?) -> ([Number] -> [Number])
+    var f = R.composeWith(function(f, res) {
+      return f(res);
+    })([R.map, R.multiply, parseInt]);
+
+    eq(f.length, 0);
+    eq(f('10')([1, 2, 3]), [10, 20, 30]);
+    eq(f('10', 2)([1, 2, 3]), [2, 4, 6]);
+  });
+
+  it('performs right-to-left function while not nil result', function() {
+    var isOdd = R.modulo(R.__, 2);
```

Comment:
Please use `R.flip(R.modulo)(2)`; I'm right now working on removing the placeholder.